### PR TITLE
Fixed the SubprocessLauncher missing app_custom_folder in PYTHONPATH

### PR DESCRIPTION
**Pull request by yhwen**
…ath.

Fixes # .

### Description

Fixed the SubprocessLauncher missing app_custom_folder in the PYTHONPATH, which caused the ScriptRunner launch_external_process failure.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/app/simulator/simulator_runner.py [hunk 1]
```diff
@@ -702,7 +703,7 @@ def do_one_task(self, client, num_of_threads, gpu, lock, timeout=60.0, task_name
         if gpu:
             command += " --gpu " + str(gpu)
         new_env = os.environ.copy()
-        new_env["PYTHONPATH"] = os.pathsep.join(self._get_new_sys_path()) + os.pathsep + app_custom_folder
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/app/simulator/simulator_runner.py:705]: should we just remove "_get_new_sys_path" if it is not used anymore?

File: nvflare/private/fed/utils/fed_utils.py [hunk 2]
```diff
@@ -391,3 +391,11 @@ def get_return_code(process, job_id, workspace, logger):
 
 def get_simulator_app_root(simulator_root, site_name):
     return os.path.join(simulator_root, site_name, SimulatorConstants.JOB_NAME, "app_" + site_name)
+
+
+def add_custom_dir_to_path(app_custom_folder, new_env):
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/utils/fed_utils.py:396]: I prefer `add_custom_dir_to_python_path` for more clarity, might be good to add docstring and typehint

**Review by YuanTingHsieh — COMMENTED** [review]
Added some comments

File: nvflare/app_common/launchers/subprocess_launcher.py [hunk 3]
```diff
@@ -50,27 +52,32 @@ def __init__(self, script: str, launch_once: bool = True, clean_up_script: Optio
...
 
-    def _start_external_process(self):
+    def _start_external_process(self, fl_ctx: FLContext):
         if self._process is None:
             command = self._script
             env = os.environ.copy()
             env["CLIENT_API_TYPE"] = "EX_PROCESS_API"
 
+            workspace = fl_ctx.get_prop(FLContextKey.WORKSPACE_OBJECT)
+            job_id = fl_ctx.get_prop(FLContextKey.CURRENT_JOB_ID)
```
[the remark this answers is not in the document]
Comment (yanchengnv) [reply nvflare/app_common/launchers/subprocess_launcher.py:77]: Both work, but easier to get it from fl_ctx directly.

**Review by yanchengnv — APPROVED** [review]
LGTM

**Pull request merged**